Objective-C/CBLProperties.mm

```diff
@@ -146,97 +145,403 @@ - (BOOL) containsObjectForKey: (NSString*)key {
 
 
 - (nullable id) objectForKeyedSubscript: (NSString*)key {
-    if (_properties)
-        return _properties[key];
-    else
-        return [self fleeceValueToObject: [self fleeceValueForKey: key]];
+    id obj = _properties ? _properties[key] : nil;
+    if (!obj) {
+        auto value = [self fleeceValueForKey: key];
+        obj = [self fleeceValueToObject: value  forKey: key asJson: NO];
```

Comment:
You can check for `value==nullptr` here and return nil, since it's a common case, and the other two calls are just no-ops.